docs/dev/docker.rst

```diff
@@ -0,0 +1,51 @@
+.. _docker:
+
+Builds and Dependency Management
+================================
+
+General Structure of Programming Exercise Execution
+---------------------------------------------------
+
+Artemis uses docker containers to run programming exercises. This ensures that the students' code does not have direct access to the build agents' hardware.
+To reduce the time required for each test run, these docker containers already include commonly used dependencies such as JUnit.
+
+There are currently docker containers for:
+
+- `Maven (Java and Kotlin) <https://github.com/ls1intum/artemis-maven-docker>`_
+- `OCaml <https://github.com/ls1intum/artemis-ocaml-docker>`_
+- `Python <https://github.com/ls1intum/artemis-python-docker>`_
+- `C <https://github.com/ls1intum/artemis-c-docker>`_
+- `Haskell <https://github.com/lukasstevens/docker-fpv-stack>`_ (external)
+- `Swift <https://github.com/norio-nomura/docker-swiftlint>`_ (external)
+- `Assembler <https://hub.docker.com/r/tizianleonhardt/era-artemis-assembler>`_ (external)
+- `VHDL <https://hub.docker.com/r/tizianleonhardt/era-artemis-vhdl>`_ (external)
+
+Steps for Updating the DockerBuilds used in Artemis
+---------------------------------------------------
+Updating can only be done with access to the account on Dockerhub. If you need access, contact `Stephan Krusche <krusche@in.tum.de>`_.
+
+1. Update the dependencies via a pull request in the docker container. Each docker container has its own repository as listed above
+2. After the PR got merged, wait for the "latest" build on Dockerhub to finish
+
+   - Choose the correct repository on `Dockerhub <https://hub.docker.com/orgs/ls1tum/repositories>`_
+   - Go to the "Builds" tab of that repository
+3. Wait for a successful build of "latest"
```

Comment:
This appears to be redundant to 2., if it refers to the same build it would maybe suit better as third subitem of 2., if not please clarify what other build the user should wait for